AdaptorsFunctionality/src/au/gov/asd/tac/constellation/functionality/adaptors/dataaccess/plugins/extend/ExtendFromGDELTPlugin.java

```diff
@@ -0,0 +1,149 @@
+import au.gov.asd.tac.constellation.plugins.parameters.types.IntegerParameterType;
+import au.gov.asd.tac.constellation.plugins.parameters.types.LocalDateParameterType;
+import au.gov.asd.tac.constellation.plugins.parameters.types.MultiChoiceParameterType;
+import au.gov.asd.tac.constellation.plugins.parameters.types.MultiChoiceParameterType.MultiChoiceParameterValue;
+import au.gov.asd.tac.constellation.views.dataaccess.DataAccessPlugin;
+import au.gov.asd.tac.constellation.views.dataaccess.DataAccessPluginCoreType;
+import au.gov.asd.tac.constellation.views.dataaccess.templates.RecordStoreQueryPlugin;
+import java.io.IOException;
+import java.time.LocalDate;
+import java.time.Month;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.openide.util.Exceptions;
+import org.openide.util.NbBundle.Messages;
+import org.openide.util.lookup.ServiceProvider;
+import org.openide.util.lookup.ServiceProviders;
+
+/**
+ * RRead graph data from a GDELT file and add it to a graph.
+ *
+ * @author canis_majoris
+ */
+@ServiceProviders({
+    @ServiceProvider(service = DataAccessPlugin.class),
+    @ServiceProvider(service = Plugin.class)})
+@PluginInfo(pluginType = PluginType.IMPORT, tags = {"IMPORT", "EXTEND"})
+@Messages("ExtendFromGDELTPlugin=Extend From GDELT Knowledge Graph")
+public class ExtendFromGDELTPlugin extends RecordStoreQueryPlugin implements DataAccessPlugin {
+
+    // plugin parameters
+    public static final String LOCAL_DATE_PARAMETER_ID = PluginParameter.buildId(ExtendFromGDELTPlugin.class, "local_date");
+    public static final String CHOICE_PARAMETER_ID = PluginParameter.buildId(ExtendFromGDELTPlugin.class, "choice");
+    public static final String LIMIT_PARAMETER_ID = PluginParameter.buildId(ExtendFromGDELTPlugin.class, "limit");
+
+    @Override
+    public String getType() {
+        return DataAccessPluginCoreType.EXTEND;
+    }
+
+    @Override
+    public int getPosition() {
+        return 500;
+    }
+
+    @Override
+    public String getDescription() {
+        return "Extend on selected graph entities to import from GDELT";
+    }
+    
+    @Override
+    public PluginParameters createParameters() {
+        final PluginParameters params = new PluginParameters();
+
+        /**
+         * The date to read from
+         */
+        final PluginParameter<LocalDateParameterType.LocalDateParameterValue> date = LocalDateParameterType.build(LOCAL_DATE_PARAMETER_ID);
+        date.setName("Date");
+        date.setDescription("Pick a day to import");
+        date.setLocalDateValue(LocalDate.of(2020, Month.JANUARY, 1));
+        params.addParameter(date);
+        
+        final PluginParameter<MultiChoiceParameterValue> choices = MultiChoiceParameterType.build(CHOICE_PARAMETER_ID);
+        choices.setName("Relationship Options");
+        choices.setDescription("Choose which relationship types to hop on");
+        MultiChoiceParameterType.setOptions(choices, Arrays.asList(

```

Comment:
It might be worth constructing this list around an enum, to make it easier to manage changes in the future.